Cargo.toml

```diff
@@ -82,6 +82,7 @@ timezone_provider = { workspace = true, optional = true }
 # System time feature
 web-time = { workspace = true, optional = true }
 iana-time-zone = { workspace = true, optional = true }
+bnum = "0.13.0"
```

Comment:
issue: hmm, I know this was @nekevss's suggestion, but I'm not really excited about adding a new dependency just for an optimization, especially one we haven't measured the benefit of)

suggestion: Can we implement one of the earlier suggestions to use narrow stdlib integer types and flatten Duration, and then have a separate discussion for adding the `bnum` dependency? I imagine that switchover will be cleaner.


@nekevss opinions? 


(Given that this is reviewed already, I wouldn't recommend changing the PR until we decide what we want here, it's possible temporal_rs decides to take on the bnum dep)

Reply:
Honestly, the one thing I've been most iffy about is adding the extra dependency (to the point of debating looking up how to manually implement arbitrarily sized integers 😅)

Hmmmmmm, I can flatten the struct and do some of the specification updates in a separate PR. If I realized that was going to be such a big portion of this PR, then I would've wanted that split off in a different PR anyways. My general concern is that the resulting Duration type is going to be huge.

Reply:
For a stack type, huge isn't really a big deal most of the time.

For JS implementors it will likely be heap allocated. But it's not like people are making thousands of these, I think.